### PR TITLE
[ntuple] Several fixes for RNTuple's minifile

### LINKED ISSUE #10559: [ntuple] Fix low-level TFile embedding
RNTuple's minifile writer seems to create subtly corrupted ROOT files. If such a file is opened by `TFile` in update mode and reopened for reading, the map reads like

```
20220511/094820  At:100       N=104       TFile                     
20320223/050248  At:204       N=68423                    CX = -11523.26 
20531200/160357  At:68627     N=14215937                 CX = -24.78 
20531200/160357  At:14284564  N=1         END           
```

On first sight, the file remains usable/readable though.

### PR DESCRIPTION
**Pull request by jblomer**
Fixes #10559

### CODE REVIEW

File: tree/ntuple/v7/src/RMiniFile.cxx [hunk 1]
```diff
@@ -1126,6 +1165,7 @@ std::uint64_t ROOT::Experimental::Internal::RNTupleFileWriter::RFileProper::Writ
    RTFString strObject;
    RTFString strTitle;
    RTFKey keyHeader(offset, offset, strClass, strObject, strTitle, len, nbytes);
+   keyHeader.MakeBigKey();
```
Comment (pcanal) [inline tree/ntuple/v7/src/RMiniFile.cxx:1168]: Consider adding a comment why this is a unconditional.

Comment (pcanal) [reply tree/ntuple/v7/src/RMiniFile.cxx:1171]: Ok.  Can  you remind me why we don't know whether the key will land past the 2GB line (TKey seems to be able to know)?

**Review by jalopezg-r00t — COMMENTED** [review]
LGTM (as in I don't see anything suspicious), but I must defer the approval to @pcanal here.

File: tree/ntuple/v7/src/RMiniFile.cxx (hunk 1 above)
Comment (jblomer) [reply tree/ntuple/v7/src/RMiniFile.cxx:1171]: We use `TKey::Create()` to reserve the space for the key in the file. To find the space, `Create()` already needs to know the length of the key and the big key provides a key length that always works. I think `TBasket` does the same, i.e. it is unconditionally a big key.

Comment (pcanal) [reply tree/ntuple/v7/src/RMiniFile.cxx:1171]: You are completely right. I had forgotten https://github.com/root-project/root/commit/3970c0beada6d13d4087dd7ac34def7aba390887.

**Review by pcanal — APPROVED** [review]

**Pull request merged**